sklearn/linear_model/base.py

```diff
@@ -436,14 +436,14 @@ def fit(self, X, y, sample_weight=None):
                 sample_weight).ndim > 1):
             sample_weight = column_or_1d(sample_weight, warn=True)
 
-        X, y, X_mean, y_mean, X_std = self._center_data(
-            X, y, self.fit_intercept, self.normalize, self.copy_X,
-            sample_weight=sample_weight)
-
         if sample_weight is not None:
             # Sample weight can be implemented via a simple rescaling.
             X, y = _rescale_data(X, y, sample_weight)
 
+        X, y, X_mean, y_mean, X_std = self._center_data(
+            X, y, self.fit_intercept, self.normalize, self.copy_X,
+            sample_weight=None)
```

Comment:
how to update sci kit to get this bug fixed?

On Thu, Oct 22, 2015 at 9:56 AM, Alexandre Gramfort <
notifications@github.com> wrote:

> In sklearn/linear_model/base.py
> <https://github.com/scikit-learn/scikit-learn/pull/5526#discussion_r42750244>
> :
>
> >          if sample_weight is not None:
> >              # Sample weight can be implemented via a simple rescaling.
> >              X, y = _rescale_data(X, y, sample_weight)
> >
> > +        X, y, X_mean, y_mean, X_std = self._center_data(
> > +            X, y, self.fit_intercept, self.normalize, self.copy_X,
> > +            sample_weight=None)
>
> or I saw #5357 (comment)
> <https://github.com/scikit-learn/scikit-learn/pull/5357#issuecomment-150180394>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/5526/files#r42750244>.
>